lib/gh/repo.go

```
@@ -95,6 +96,20 @@ func (t Token) NewClient() (*github.Client, context.Context) {
 	return github.NewClient(retryClient.StandardClient()), ctx
 }
 
+func (t Token) NewGraphQLClient() (*githubv4.Client, context.Context) {
+	ctx := context.Background()
+
+	if t.Token == nil {
+		// TODO do we want to error here? `NewClient()` above does not, but requests made by this client will fail without a token
```

if this specific API/client requires a token to do anything then yes we should return an error